pkg/pgmodel/extension.go

```diff
@@ -176,27 +182,27 @@ func fetchInstalledExtensionVersion(conn *pgx.Conn, extensionName string) (semve
 	return v, true, nil
 }
 
-func correctVersionString(v string) string {
+func correctVersionString(v string, extName string) string {
 	//we originally published the extension as "0.1" which isn't a valid semver
-	if v == "0.1" {
+	if extName == "promscale" && v == "0.1" {
 		return "0.1.0"
 	}
 	return v
 }
 
-func getSqlVersion(v semver.Version) string {
-	if v.String() == "0.1.0" {
+func getSqlVersion(v semver.Version, extName string) string {
+	if extName == "promscale" && v.String() == "0.1.0" {
 		return "0.1"
 	}
 	return v.String()
 }
 
-// getNewExtensionVersion returns the highest version allowed by ExtVersionRange
-func getNewExtensionVersion(availableVersions semver.Versions) (semver.Version, bool) {
+// getNewExtensionVersion returns the highest version allowed by validRange
+func getNewExtensionVersion(availableVersions semver.Versions, validRange semver.Range) (semver.Version, bool) {
```

Comment:
Should we rename this to `getNewVersionWithinRange()`?

Reply:
I don't think so since there is no other function that doesn't take a range